### PR TITLE
fix(enum): deprecates Party Type enum (#1023)

### PR DESCRIPTION
**Pull request by KnownBlackHat**
## Summary

Deprecates  the `PartyType` enum 

- Close GH-1023

## Checklist

- [X] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [X] I have formatted the code properly by running `pdm lint`
    - [X] I have type-checked the code by running `pdm pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by Victorsitou — CHANGES_REQUESTED** [review]
The enum should be deprecated rather than removed to avoid breaking changes

**Comment by KnownBlackHat** [comment]
> 

yep doing that rn

File: disnake/abc.py [hunk 1]
```diff
@@ -1287,6 +1287,11 @@ async def create_invite(
         :class:`.Invite`
             The newly created invite.
         """
+        if isinstance(target_application, PartyType):
+            utils.warn_deprecated(
+                "target_application parameter is deprecated and will be removed in future version",
+                stacklevel=2,
+            )
```
Comment (shiftinv) [inline disnake/abc.py:1294]: `target_application` is unlikely to be removed, it's still a valid and documented API field.
It should however now support `Object`s as an alternative (with accompanying changelog and `versionchanged` documentation), and `PartyType` for backwards compatibility until the next version

File: changelog/1035.deprecate.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:attr:`enum.PartyType` is obsolete.
+
+The PartyType enum contains IDs of voice activities, which aren't documented in any official capacity. New activities get added/removed constantly, so keeping up with maintaining that list is rather difficult.
+
+Since the introduction of this enum, activities have been released officially.
+If anything, providing activities should really be more of a bot's responsibility, not the library's, so we should consider deprecating and removing this enum.
```
Comment (shiftinv) [inline changelog/1035.deprecate.rst:6]:
```suggestion
The :class:`PartyType` enum is deprecated, as voice activity IDs are not officially documented and the enum regularly becomes outdated.
```
It doesn't have to be too wordy, and multiline changelogs don't render that well:
image

**Review by shiftinv — COMMENTED** [review]
Could you also add a deprecation note (`.. deprecated:: 2.9` etc.) below here, similar to other deprecation notes? https://github.com/DisnakeDev/disnake/blob/473bafbbdda4185696d29033b4f3ba05e0126757/docs/api/voice.rst?plain=1#L111-L113

File: disnake/abc.py (hunk 1 above)
Comment (shiftinv) [reply disnake/abc.py]: This doesn't seem to be resolved yet (and neither is the comment above)?

At least for this comment, something like [this](https://gist.github.com/shiftinv/e8f091e813e5a598af10e0c01e538a08) could work, I believe.

**Review by Victorsitou — APPROVED** [review]

**Review by shiftinv — APPROVED** [review]
